### PR TITLE
Fixing ModelOutput training

### PR DESCRIPTION
**Pull request by svlandeg**
Follow-up for https://github.com/explosion/spacy-transformers/pull/283

When inspecting the model training with the `ModelOutput` changes, `unsplit_by_doc` received a different dimension for the `arrays` input parameter: basically the `pooler_output` tensor would be missing. I think the `backprop` in `_convert_transformer_outputs` should include all tensors - not just the last hidden state.

To make that work, the `d_model_output` object holding the gradients in `backprop_trf_to_tensor` should have the same keys as the original object.

In all of this, `"attentions"` should be excluded though. It feels kind of bad to hardcode this, but it is a specifically defined field in [`BaseModelOutput`](https://huggingface.co/transformers/main_classes/output.html#basemodeloutput) so I think it's fine?

I haven't yet tested this on a serious training dataset though, so I'll keep this in draft until we've had a chance to test further 🤞

### CODE REVIEW

File: spacy_transformers/data_classes.py [hunk 1]
```diff
@@ -177,8 +180,10 @@ def zeros(cls, length: int, width: int, *, xp=numpy) -> "TransformerData":
         )
 
     @property
-    def tensors(self) -> Tuple[Union[FloatsXd, List[FloatsXd]]]:
-        return self.model_output.to_tuple()
+    def tensors(self, exlude=EXCLUDED_KEYS) -> Tuple[Union[FloatsXd, List[FloatsXd]]]:
+        return tuple(
+            self.model_output[k] for k in self.model_output.keys() if k not in exlude
+        )
```
Comment (svlandeg) [inline spacy_transformers/data_classes.py:186]: This particular change isn't strictly necessary, but I thought it would keep the API consistent (same comment for similar change in the same file)

File: spacy_transformers/layers/transformer_model.py [hunk 2]
```diff
@@ -216,10 +214,11 @@ def _convert_transformer_inputs(model, wps: WordpieceBatch, is_train):
 
 def _convert_transformer_outputs(model, inputs_outputs, is_train):
     _, model_output = inputs_outputs
+    tensors = tuple(model_output[k] for k in model_output.keys() if k not in EXCLUDED_KEYS)
 
     def backprop(d_model_output: ModelOutput) -> ArgsKwargs:
         return ArgsKwargs(
-            args=(model_output.last_hidden_state,),
+            args=(tensors,),
```
Comment (svlandeg) [inline spacy_transformers/layers/transformer_model.py:221]: This is the main part where I found divergence between the current code and before #283: before, the `tensors` would include more than just the last hidden state, e.g. for `roberta-base` there would also be a `pooler_output` output that would get gradients too.

**Comment by svlandeg** [comment]
This PR is hopefully not needed anymore after https://github.com/explosion/spacy-transformers/pull/301, but I want to do some additional checks before I close this.
